core/tests/protractor_utils/users.js

```diff
@@ -49,14 +49,12 @@ var _completeSignup = function(username) {
   browser.get('/signup?return_url=http%3A%2F%2Flocalhost%3A9001%2F');
   element(by.css('.protractor-test-username-input')).sendKeys(username);
   element(by.css('.protractor-test-agree-to-terms-checkbox')).click();
-  element(by.css('.protractor-test-register-user')).click();
+  return element(by.css('.protractor-test-register-user')).click();
 };
 
 var createUser = function(email, username) {
   login(email);
-  _completeSignup(username);
-  general.waitForSystem();
-  logout();
+  _completeSignup(username).then(logout());
```

Comment:
I don't think we should be turning stuff into promises where not needed. Protractor is meant to try and handle the async stuff behind the scenes...

Ditto below. ExpectedConditions would be better if you're actually waiting for something IMO.

Reply:
I think ExpectedConditions also return a promise which needs to be resolved. The `_completeSignup()` function originally was:
```js
var _completeSignup = function(username) {
  browser.get('/signup?return_url=http%3A%2F%2Flocalhost%3A9001%2F');
  element(by.css('.protractor-test-username-input')).sendKeys(username);
  element(by.css('.protractor-test-agree-to-terms-checkbox')).click();
  element(by.css('.protractor-test-register-user')).click();
};
```
Each of the `.click()` operations return a promise. We seemed to be waiting for that promise to resolve originally. I returned that promise from the function `_completeSignup` and added `then`, which waits until it is resolved before the next statement is executed...